pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -216,41 +244,52 @@ func getPortMap(pod *v1.Pod) map[string]uint32 {
 }
 
 // deleteIP returns true if the pod and ip are really deleted.
-func (pc *PodCache) deleteIP(ip string, podKey types.NamespacedName) bool {
+func (pc *PodCache) deleteIPs(ips []string, podKey types.NamespacedName) bool {
 	pc.Lock()
 	defer pc.Unlock()
-	if pc.podsByIP[ip] == podKey {
-		delete(pc.podsByIP, ip)
-		delete(pc.IPByPods, podKey)
-		return true
+	var deleteIPFlag bool
+	for _, itemIP := range ips {
+		if pc.podsByIP[itemIP] == podKey {
+			delete(pc.podsByIP, itemIP)
+			delete(pc.IPByPods, podKey)
+			deleteIPFlag = true
+		} else {
+			deleteIPFlag = false
+		}
 	}
-	return false
+	return deleteIPFlag
```

Comment:
I don't think this works, we will incorrectly behave if we only delete one IP

Reply:
okay, I change the logic for `deleteIPs` function.